### ISSUE TITLE
Show the Introductory Popup / Banner to view-only users and secondary admins

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#8172]
## Feature Description

Show the Introductory Popup / Banner to view-only users and secondary admins to introduce the feature once it has been set up.

See [dashboard sharing](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.w0br3nfp3mbe) and [secondary admins](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.gkj45q2rrrne) in the design doc.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Given that the Audience Segmentation feature has been set up by the primary admin (i.e. an admin has enabled the feature via the Setup CTA Banner, or the Setup CTA on the Settings screen):
  * When a secondary admin, or a view-only user subsequently visits the main dashboard, the Introductory Popup / Banner should be displayed (see https://github.com/google/site-kit-wp/issues/8171).
  * Clicking on the Popup / Banner's **Show me** CTA will scroll to the Audiences Widget Area, and permanently dismiss the Popup / Banner.
  * Clicking on the Popup / Banner's **Got it** CTA will permanently dismiss the Popup / Banner.
  * The Popup / Banner should not display on the entity dashboard.

## Implementation Brief

- [x] Rendering the banner/popup is handled [here](https://github.com/google/site-kit-wp/blob/de09a10c23b15d090ed2207693f6ac9e1bb67f7e/assets/js/components/OverlayNotification/OverlayNotificationsRenderer.js#L38-L40). It will be visible when the Audience Segmentation feature is enabled.
  - [x] Update the condition to also check if the Audience Segmentation feature has already been setup. This can be checked with the flag `audienceSegmentationSetupCompleted` in Analytics settings being introduced in #8130.
- [x] In `assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationIntroductoryOverlayNotification.js`
  - [x] Import `getContextScrollTop` from `assets/js/util/scroll.js`.
  - [x] Add a new handler `scrollToWidgetAndDismissNotification` for scrolling and dismissing the notification.
    - [x] This function should call the `dismissNotification`.
    - [x] Pass `context` as  `googlesitekit-widget-area--mainDashboardTrafficPrimary` and `breakpoint` using `useBreakpoint` hook to `getContextScrollTop`.

- [x] As the `getContextScrollTop` is now being used to get the position for widgets and widget areas as well as widget context elements, rename this function to `getNavigationalScrollTop` and update all its usages indicating it's a general purpose utility for use in scrolling to an area which is within the remit of the dashboard navigation, below the header.

### Test Coverage

* Add a test for `AudienceSegmentationIntroductoryOverlayNotification` component to test the scroll behaviour.
* Add a test for `AudienceSegmentationIntroductoryOverlayNotification` component to test the dismissal behaviour.
* Update story for `AudienceSegmentationIntroductoryOverlayNotification` to include a new scenario dependent on `audienceSegmentationSetupCompleted` flag.

### For secondary admins.

1. Enable the audience segmentation feature and complete the setup with primary admin account.

2. Create an another admin account and login with that account. Pop up notification must be visible for the secondary admin user.

### For view only user

1. Enable the audience segmentation feature and complete the setup with primary admin account.

2. Create a view only user account and provide it the view only access using `Sharing Settings` in Site Kit. Note that we will need to provide view only access to Analytics module so that view-only user has access to the Audience Segmentation feature.

3. Login with view only user's account and go to Site Kit dashboard, there should be the pop-up notification appearing for the user.

## Changelog entry

* Ensure the Audience Segmentation Introductory Popup / Banner is only shown to to view-only users and secondary admins once the feature has been setup by the initial admin.

**Comment by techanvil** [comment]
Hi @ankitrox, thanks for drafting this IB. Here are a few points:

> - [ ] Rendering the banner/popup is already been taken care [here](https://github.com/google/site-kit-wp/blob/de09a10c23b15d090ed2207693f6ac9e1bb67f7e/assets/js/components/OverlayNotification/OverlayNotificationsRenderer.js#L38-L40). It will be visible when the audience segmentation feature is enabled.

- Actually, while this is true in itself, the banner should not be shown purely based on whether the feature flag is enabled. In fact the banner should only be shown under the conditions described in the AC, i.e. once the feature has been setup by a primary admin. I imagine we can add a check for `audienceSegmentationSetupComplete` (being introduced via https://github.com/google/site-kit-wp/issues/8130) in the conditions for rendering the banner (either in `OverlayNotificationsRenderer`, or to potentially return `null` in `AudienceSegmentationIntroductoryOverlayNotification`).
- Further to the above I've added #8130 as a dependency to this issue.

---


- As to the rest of the IB - there's no need to add a `scrollToAudienceWidgetArea` parameter and modify `dismissNotification`. Instead, we should simply follow the example of `AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification` and provide a separate click handler for the **Show me** CTA with a similar implementation:

https://github.com/google/site-kit-wp/blob/51452f6f0d6e3afedfbfc841c5649d328be6ca0f/assets/js/components/OverlayNotification/AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification.js#L182-L201

- We should pass `.googlesitekit-widget-area--mainDashboardTrafficAudienceSegmentation` rather than `googlesitekit-widget-area--mainDashboardTrafficPrimary` to `getContextScrollTop()` to ensure we scroll to the Audiences Widget Area rather than the top of the Traffic context.
- Incidentally seeing as `getContextScrollTop()` is now being used to get the position for widgets and widget areas as well as widget context elements, we should rename the function. I'd suggest a new name of `getNavigationalScrollTop()` indicating it's a general purpose utility for use in scrolling to an area which is within the remit of the dashboard navigation, below the header.

**Comment by ankitrox** [comment]
Hi @techanvil ,

Thanks for reviewing the IB. I've addressed the feedback points mentioned in the comments and updated the IB.

Assigning this to you for re-review.

**Comment by techanvil** [comment]
Thanks @ankitrox. The main body of the IB is looking good (although it needed a tweak for formatting, please review). I have though realised the Test Coverage section is written in a way that would encourage more spying on implementation details.

We don't really want to test whether `getContextScrollTop()` has been called as that's not verifying the actual behaviour of the component. If anything a spy and assertion on `global.scrollTo()` would be more appropriate here, that being the public DOM API call that would be the clearest indication we've actually scrolled or not.

Also, although it does look like we're in the process of setting a precedent of testing whether an action has been dispatched by spying on it, we should still try to stick with our more usual approach where we can. As such the more conventional approach would be to mock the `dismiss-item` endpoint and assert the DOM after interacting with the component.

For example here are a couple of existing tests for dismissing a UI item:

https://github.com/google/site-kit-wp/blob/817317dfc6e20c72f3637bf823db540f247af6bc/assets/js/components/OverlayNotification/AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification.test.js#L436-L466

https://github.com/google/site-kit-wp/blob/817317dfc6e20c72f3637bf823db540f247af6bc/assets/js/modules/adsense/components/dashboard/AdSenseConnectCTAWidget.test.js#L113-L125

To be honest, we don't usually specify the test coverage in an IB in such detail. I would suggest simply specifying that we provide test coverage for the relevant behaviour, and leave the details to implementation time.

**Comment by ankitrox** [comment]
Hi @techanvil 
Thanks for the feedback. I agree that details in the test coverage section is not required, so I updated the section just to mention which tests will be required to cover the feature.

Assigning this to you to re-review the test coverage section.

Thanks again!

**Comment by techanvil** [comment]
That's great, thanks @ankitrox. The IB LGTM! :white_check_mark:

**Comment by ankitrox** [comment]
The PR #9239 is ready, but this issue will be parked in execution till #8130 gets merged. Once #8130 is merged, need to do following checks before moving it to CR.

1. Merge develop into feature branch and resolve any conflicts.
2. Make sure all tests and VRTs are passing.
3. Test the feature on local.
4. Write QAB for the issue.

**Comment by techanvil** [comment]
Hi @ankitrox, the QAB for this will also need an update as the popup should be visible for secondary admins as well as view-only users.

**Comment by techanvil** [comment]
Hi @ankitrox, thanks for updating the QAB. One minor point, for the view-only case, it would be helpful to explain that it's the Analytics module that needs to be shared in order for the view-only user to have access to the Audience Segmentation feature.

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

- Tested on dev environment.
- Verified on both 'view only dashboard' and 'secondary admin dashboard.'
- Verified that introductory Popup/Banner appears to view only users and secondary admins.


@ankitrox 

**Question** -I noticed that if I log in as a secondary admin and go directly to the main dashboard without visiting the 'View Only Dashboard,' I receive an insufficient permission error for the Audience widget. However, if I either disconnect and visit the 'View Only Dashboard' first or access the 'View Only Dashboard' before signing in, I don’t encounter the insufficient permission error.

Note: All admins have view-only access to Analytics.

Should I create a separate issue for this ?

<details>

![image](https://github.com/user-attachments/assets/c68ebcae-54c0-41aa-ae32-a02a0177f511)


https://github.com/user-attachments/assets/438bbed6-474a-4fa7-a757-ec6bdef084b9

</details>

**- View only dashboard-**

<details>


https://github.com/user-attachments/assets/fc22b7c2-cdca-4ae5-a6ec-a21441d6214d

</details>

**Comment by ankitrox** [comment]
Thank you @mohitwp 

I've created a follow-up PR #9416 to address the issue you have reported.

@techanvil Can you please look review the PR if you have bandwidth? I have not assigned you directly in case you are already occupied with something else.

**Comment by techanvil** [comment]
Thanks @ankitrox! I'll get that PR reviewed :+1:

**Comment by techanvil** [comment]
Reviewed and merged, back to you for another pass @mohitwp.

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

- Tested on dev environment.
- Verified that now on 'entity dashboard' popup is not appearing.

<details>


https://github.com/user-attachments/assets/35745bc2-5183-4fa8-8214-5a97b67727e2

</details>

@ankitrox 

_**Question**_ 
_When visiting the 'view only' dashboard, I'm encountering a console error:` Cannot read properties of undefined (reading 'error'). `Could you please confirm if this is related to the changes introduced in this ticket?_

<details>

![image](https://github.com/user-attachments/assets/4e90ff7d-09df-488f-87a9-7a0d3456821c)

![image](https://github.com/user-attachments/assets/7012def8-66fb-44cd-b720-4d61704819bb)

</details>

**Comment by techanvil** [comment]
Hi @mohitwp, thanks for spotting that. It does not appear to be related to the changes in this issue, please can you raise a new one for it?

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on dev environment.
- Verified on both 'view only dashboard' and 'secondary admin dashboard.'
- Verified that introductory Popup/Banner appears to view only users and secondary admins.
- Verified that now on 'entity dashboard' popup is not appearing.


<details>


https://github.com/user-attachments/assets/35745bc2-5183-4fa8-8214-5a97b67727e2

</details>
